kactus/main.go

```diff
@@ -564,29 +565,31 @@ func (cc *cniContext) getResourceMap(no *netObject, resourceMap map[string]*Reso
 	// Get resourceName annotation from the Network CR
 	deviceID := ""
 	resourceName, ok := no.GetAnnotations()[resourceNameAnnot]
-	if ok && cc.pod.Name != "" && cc.pod.Namespace != "" {
-		// ResourceName annotation is found; try to get device info from resourceMap
-		kc.LogDebug("getResourceMap: found resourceName annotation : %s\n", resourceName)
-
-		if resourceMap == nil {
-			ck, err := GetResourceClient()
-			if err != nil {
-				return nil, deviceID, resourceName, fmt.Errorf("getResourceMap: failed to get a ResourceClient instance: %v", err)
-			}
-			resourceMap, err = ck.GetPodResourceMap(cc.pod)
-			if err != nil {
-				return resourceMap, deviceID, resourceName, fmt.Errorf("getResourceMap: failed to get resourceMap from ResourceClient: %v", err)
-			}
-			kc.LogDebug("getResourceMap: resourceMap instance: %+v\n", resourceMap)
+	if !ok || cc.pod == nil || cc.pod.Name == "" || cc.pod.Namespace == "" {
+		return resourceMap, deviceID, resourceName, nil
```

Comment:
It depends what the previous intended behavior was.
I think the intended behavior was to simply skip the logic if there weren't a Pod defined. I think that's why there was a check for the Pod name and the Pod namespace. In that case, we don't need to send an error AND we also don't want to return nil for the resourceMap because we updating that map later on. (Like a loop that updates the resourceMap kinda thing)

```go
    for i, podNet := range networks {                                                         
        if i != 0 {                                                              
            netConf.WriteString(",")                                                          
        }                                                                                     
                                                                                 
        primary := false                                                         
        if !cc.auxNetOnly && podNet.IsPrimary {                                  
            primary = true                                                       
        }                                                                            
                                                                                  
        nc, updatedResourceMap, err := cc.getDelegateNetConf(podNet.NetworkName, resourceMap,
        if err != nil {                                                                     
            return "", fmt.Errorf("Kactus: failed getting the netplugin: %v", err)
        }                                                                        
        resourceMap = updatedResourceMap                                                      
        netConf.WriteString(nc)                                                  
    }      
```

If the intended behavior was to return an error (which the previous code doesn't do), then yes we could return an error and nil for the resourceMap.

What do you think?

Reply:
Ok see what you mean with regards to previous code. Let's go for that (it's equivalent to previous, with extra check).